BSP/STM32F413/Src/BSP_SPI.c

```diff
@@ -1,8 +1,28 @@
+/* Copyright (c) 2020 UT Longhorn Racing Solar */
+
 #include "BSP_SPI.h"
 #include "stm32f4xx.h"
+#include "os.h"
+#include "BSP_OS.h"
+
+static bsp_os_t *os;
+static bsp_os_t *os3;
 
 // Use this macro function to wait until SPI communication is complete
+#ifdef BAREMETAL
 #define SPI_Wait(SPIx)		while(((SPIx)->SR & (SPI_SR_TXE | SPI_SR_RXNE)) == 0 || ((SPIx)->SR & SPI_SR_BSY))
+#endif
+
+#ifdef RTOS
+#define SPI_Wait(SPIx)		if(((SPIx)->SR & (SPI_SR_TXE | SPI_SR_RXNE)) == 0 || ((SPIx)->SR & SPI_SR_BSY)){ \
+								if(SPIx == SPI1){	\
+									os->pend();	\
+								}	\	
+								else if(SPIx == SPI3){	\
+									os3->pend();	\
+								}	\
+							} 
+#endif
```

Comment:
Maybe instead of `#define`s, we can use these things called `functions` where we don't have to break lines with `\`. If you really want a minor speed optimization, just declare them `inline`.